contrib/crc32c-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,123 @@
+set(CRC32C_DIR ${ClickHouse_SOURCE_DIR}/contrib/crc32c)
+
+if(CRC32C_USE_GLOG)
+  # glog requires this setting to avoid using dynamic_cast.
+  set(DISABLE_RTTI ON CACHE BOOL "" FORCE)
+
+  # glog's test targets trigger deprecation warnings, and compiling them burns
+  # CPU cycles on the CI.
+  set(BUILD_TESTING_SAVED "${BUILD_TESTING}")
+  set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
+  add_subdirectory("third_party/glog" EXCLUDE_FROM_ALL)
+  set(BUILD_TESTING "${BUILD_TESTING_SAVED}" CACHE BOOL "" FORCE)
+
+  # glog triggers deprecation warnings on OSX.
+  # https://github.com/google/glog/issues/185
+  if(CRC32C_HAVE_NO_DEPRECATED)
+    set_property(TARGET glog APPEND PROPERTY COMPILE_OPTIONS -Wno-deprecated)
+  endif(CRC32C_HAVE_NO_DEPRECATED)
+
+  # glog triggers sign comparison warnings on gcc.
+  if(CRC32C_HAVE_NO_SIGN_COMPARE)
+    set_property(TARGET glog APPEND PROPERTY COMPILE_OPTIONS -Wno-sign-compare)
+  endif(CRC32C_HAVE_NO_SIGN_COMPARE)
+
+  # glog triggers unused parameter warnings on clang.
+  if(CRC32C_HAVE_NO_UNUSED_PARAMETER)
+    set_property(TARGET glog
+                 APPEND PROPERTY COMPILE_OPTIONS -Wno-unused-parameter)
+  endif(CRC32C_HAVE_NO_UNUSED_PARAMETER)
+
+  set(CRC32C_TESTS_BUILT_WITH_GLOG 1)
+endif(CRC32C_USE_GLOG)
+
+configure_file(
+  "${CRC32C_DIR}/src/crc32c_config.h.in"
+  "${PROJECT_BINARY_DIR}/include/crc32c/crc32c_config.h"
+)
+
+include_directories("${PROJECT_BINARY_DIR}/include")
```

Comment:
Does `crc32c_config.h` required by other public targets or only by the crc32c itself?

I guess `include_directories` will add it to `PUBLIC` list, which will add it also to all clickhouse targets, but maybe it is enough to have it in `PRIVATE`?